modules/c++/mem/tests/ScratchVisualization.cpp

```diff
@@ -0,0 +1,477 @@
+                                                      mPrevOperations.at(ii).bytes));
+            }
+        }
+    }
+
+    /*!
+     * Runs the randomly generated test case.
+     *
+     * \param[in,out] currentOperations Operations for this iteration
+     * \param[in,out] bufferName The name/key of the new segment
+     * \param[in,out] notReleasedKeys The keys that have not been released
+     * \param[out] usedBufferSpace How much memory has already been used up
+     * \param[out] scratch Scratch memory object
+     */
+    void randomTest(std::vector<Operation>& currentOperations,
+                    unsigned char& bufferName,
+                    std::vector<unsigned char>& notReleasedKeys,
+                    size_t& usedBufferSpace,
+                    mem::ScratchMemory& scratch)
+    {
+        handlePrevOps(currentOperations, scratch);
+
+        const size_t numElements = (rand() % 150) + 20;
+        unsigned int releaseIfThree = (rand() % 3) + 1;
+
+        if ((releaseIfThree == 3) && (currentOperations.size() > 1) && !notReleasedKeys.empty())
+        {
+            unsigned int keyToReleaseIndex = (rand() % notReleasedKeys.size());
+            const std::string segmentName = std::string(1, notReleasedKeys.at(keyToReleaseIndex)) +
+                    str::toString(mIteration);
+
+            scratch.release(segmentName);
+            notReleasedKeys.erase(notReleasedKeys.begin() + keyToReleaseIndex);
+
+            currentOperations.push_back(Operation("release", segmentName, numElements));
+        }
+        else
+        {
+            const std::string segmentName = std::string(1, bufferName) + str::toString(mIteration);
+            scratch.put<sys::ubyte>(segmentName, numElements, 1, 1);
+            currentOperations.push_back(Operation("put", segmentName, numElements));
+            notReleasedKeys.push_back(bufferName);
+
+            ++bufferName;
+            usedBufferSpace += numElements;
+        }
+    }
+
+    /*!
+     * Runs a version of the unittest in which concurrent keys are released in a very specific order
+     *
+     * \param[in,out] currentOperations Operations for this iteration
+     * \param[in,out] bufferName The name/key of the new segment
+     * \param[in,out] notReleasedKeys The keys that have not been released
+     * \param[in,out] testIter Used to keep track of which step of the unittest to do
+     * \param[out] usedBufferSpace How much memory has already been used up
+     * \param[out] scratch Scratch memory object
+     */
+    void concurrentBlockTest(std::vector<Operation>& currentOperations,
+                             unsigned char& bufferName,
+                             size_t& testIter,
+                             size_t& usedBufferSpace,
+                             mem::ScratchMemory& scratch)
+    {
+        handlePrevOps(currentOperations, scratch);
+        size_t numElements = (rand() % 150) + 20;
+
+        if (testIter == 0)
```

Comment:
This looks like it is doing the same thing a `==1`. Can we change this to something like:
```
if (testIter == 0 || testIter == 1 | ...)
{
}
```